File: app/src/main/java/com/script972/clutchclient/ui/activitys/SettingActivity.java
package com.script972.clutchclient.ui.activitys;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.widget.Toolbar;
import android.text.Layout;
import android.view.View;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.script972.clutchclient.R;
import com.script972.clutchclient.core.ClutchApplication;
import com.script972.clutchclient.helpers.DialogHelper;
import com.script972.clutchclient.helpers.PrefHelper;
import com.script972.clutchclient.model.LocationPosition;


public class SettingActivity extends BaseActivity {

    //outlets
    private Toolbar tollbar;
    private TextView txtCurrentCountry;
    private LinearLayout displayLayout;
    private TextView displayCardStatus;

    private View exitLayout;
    private View locationLayout;

    private final int REQUEST_COUNTRY=312;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_setting);
        super.initCommonView();
        initView();

    }

    @Override
    protected void onResume() {
        super.onResume();

    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);

        switch (requestCode){
            case REQUEST_COUNTRY:{

                break;
            }
        }
    }

    /**
     * Method for init views
     */
    private void initView() {
        initToolbar();
        initDisplayCard();
        initLocationLayout();
        exitLayout = findViewById(R.id.exit_layout);
        exitLayout.setOnClickListener(clicker);
    }

    /**
     * init location layout
     */
    private void initLocationLayout() {
        locationLayout = findViewById(R.id.current_location_layout);
        locationLayout.setOnClickListener(clicker);
        txtCurrentCountry = findViewById(R.id.current_country_text_view);
        LocationPosition location = PrefHelper.getLocation(this);
        if(location!=null && !location.getCountry().isEmpty()) {
            txtCurrentCountry.setText(location.getCountry());
        }
    }

    /**
     * init card view display
     */
    private void initDisplayCard() {
        displayLayout=findViewById(R.id.display_layout);
        displayCardStatus = findViewById(R.id.display_card_status_text_view);
        int numberColomn = PrefHelper.getDisplayCardView(getApplicationContext());
        if(numberColomn==1) {
            displayCardStatus.setText(getResources().getString(R.string.txt_card_view_column));
        } else if (numberColomn == 2) {
            displayCardStatus.setText(getResources().getString(R.string.txt_card_view_row));
        }
        displayLayout.setOnClickListener(clicker);
    }

    /**
     * Method wich init toolbar
     */
    private void initToolbar() {
        tollbar = (Toolbar) findViewById(R.id.toolbar);
        tollbar.setNavigationIcon(getResources().getDrawable(R.drawable.ic_arrow_back));
        tollbar.setNavigationOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (validationData()){
                    saveData();
                    onBackPressed();
                }
            }
        });
        tollbar.setTitle(getResources().getString(R.string.toolbar_settings));
    }

    /**
     * Method wich check validation data
     */
    private boolean validationData() {
        return true;
    }

    /**
     * Method wich save new Settings
     */
    private void saveData() {

    }


    /**
     * Method  wich open Select Country Activity
     */
    private void openCountryLocation() {
        Intent intent=new Intent(this, SettingCountryActivity.class);
        startActivityForResult(intent,REQUEST_COUNTRY);
    }



    /**
     * Method wich display card view
     */
    private void clickDisplayCard() {
        if(PrefHelper.getDisplayCardView(getApplicationContext())==1){
            PrefHelper.setDisplayCardView(getApplicationContext(), 2);
            displayCardStatus.setText(getResources().getString(R.string.txt_card_view_row));
        } else  if(PrefHelper.getDisplayCardView(getApplicationContext()) == 2){
            PrefHelper.setDisplayCardView(getApplicationContext(), 1);
            displayCardStatus.setText(getResources().getString(R.string.txt_card_view_column));
        }
    }

    /**
     * Method for logOut
     */
    private void logOut() {
        DialogHelper.logOutDialog(getApplicationContext(), this);
    }



    //callbacks
    /**
     * General clicker
     */
    private final View.OnClickListener clicker = new View.OnClickListener() {
        @Override
        public void onClick(View v) {
            switch (v.getId()){
                case R.id.exit_layout: logOut(); break;
                case R.id.display_layout: clickDisplayCard(); break;
                case R.id.current_location_layout: openCountryLocation(); break;

            }
        }
    };




}

File: app/src/main/java/com/script972/clutchclient/ui/activities/SettingActivity.kt
package com.script972.clutchclient.ui.activities

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Toast
import com.script972.clutchclient.R
import com.script972.clutchclient.helpers.PrefHelper
import kotlinx.android.synthetic.main.activity_setting.*
import kotlinx.android.synthetic.main.toolbar.*

class SettingActivity : BaseActivity() {

    //outlets
    /*  private TextView txtCurrentCountry;
    private LinearLayout displayLayout;
    private TextView displayCardStatus;

    private View exitLayout;
    private View locationLayout;
  */

    companion object {
        val REQUEST_COUNTRY = 312
    }


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_setting)
        initView()
    }


    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        when (requestCode) {
            REQUEST_COUNTRY -> print("2")
            else -> {
                print("")
            }
        }
    }

    fun initView() {
        initToolbar()
        initDisplayCard()
        //initLocationLayout();
        //exitLayout = findViewById(R.id.exit_layout);
        //exitLayout.setOnClickListener(clicker);
    }

    /**
     * Method wich init toolbar
     */
    fun initToolbar() {
        toolbar.title = resources.getString(R.string.toolbar_settings)
        toolbar.navigationIcon = resources.getDrawable(R.drawable.ic_arrow_back)
        toolbar.setNavigationOnClickListener({
            /*if (validationData()){
                saveData();
                onBackPressed();
            }*/
            onBackPressed()
        })


    }

    fun initDisplayCard() {

        val numberColomn = PrefHelper.getDisplayCardView(applicationContext)
        if (numberColomn == 1) {
            display_card_status_text_view.text = resources.getString(R.string.txt_card_view_column)
        } else if (numberColomn == 2) {
            display_card_status_text_view.text = resources.getString(R.string.txt_card_view_row)
        }
        display_layout.setOnClickListener {
            if (PrefHelper.getDisplayCardView(applicationContext) == 1) {
                PrefHelper.setDisplayCardView(applicationContext, 2)
                display_card_status_text_view.text = resources.getString(R.string.txt_card_view_row)
            } else if (PrefHelper.getDisplayCardView(applicationContext) == 2) {
                PrefHelper.setDisplayCardView(applicationContext, 1)
                display_card_status_text_view.text = resources.getString(R.string.txt_card_view_column)
            }
        }

    }


}


/*
/**
 * Method for init views
 *//*


    */
    /**
     * init location layout
     *//*
    private void initLocationLayout() {
        locationLayout = findViewById(R.id.current_location_layout);
        locationLayout.setOnClickListener(clicker);
        txtCurrentCountry = findViewById(R.id.current_country_text_view);
        LocationPosition location = PrefHelper.getLocation(this);
        if(location!=null && !location.getCountry().isEmpty()) {
            txtCurrentCountry.setText(location.getCountry());
        }
    }

    */
    *
    /**
     * init card view display
     */



/**
 * Method wich check validation data
 *//*
    private boolean validationData() {
        return true;
    }

    */
/**
 * Method wich save new Settings
 *//*
    private void saveData() {

    }


    */
/**
 * Method  wich open Select Country Activity
 *//*
    private void openCountryLocation() {
        Intent intent=new Intent(this, SettingCountryActivity.class);
        startActivityForResult(intent,REQUEST_COUNTRY);
    }



    */

/**
 * Method for logOut
 *//*
    private void logOut() {
        DialogHelper.logOutDialog(getApplicationContext(), this);
    }



    //callbacks
    */
/**
 * General clicker
 *//*
    private final View.OnClickListener clicker = new View.OnClickListener() {
        @Override
        public void onClick(View v) {
            switch (v.getId()){
                case R.id.exit_layout: logOut(); break;
                case R.id.display_layout: clickDisplayCard(); break;
                case R.id.current_location_layout: openCountryLocation(); break;

            }
        }
    };*/